src/pages/workspace/workflows/WorkspaceAutoReportingFrequencyPage.tsx

```diff
@@ -0,0 +1,139 @@
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import MenuItem from '@components/MenuItem';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import ScreenWrapper from '@components/ScreenWrapper';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import Navigation from '@libs/Navigation/Navigation';
+import * as PolicyUtils from '@libs/PolicyUtils';
+import withPolicy from '@pages/workspace/withPolicy';
+import type {WithPolicyAndFullscreenLoadingProps} from '@pages/workspace/withPolicyAndFullscreenLoading';
+import * as Policy from '@userActions/Policy';
+import CONST from '@src/CONST';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+
+type AutoReportingFrequencyKey = Exclude<ValueOf<typeof CONST.POLICY.AUTO_REPORTING_FREQUENCIES>, 'instant'>;
+
+type WorkspaceAutoReportingFrequencyPageProps = WithPolicyAndFullscreenLoadingProps;
+
+type WorkspaceAutoReportingFrequencyPageSectionItem = {
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+};
+
+type AutoReportingFrequencyDisplayNames = Record<AutoReportingFrequencyKey, string>;
+
+const autoReportingFrequencyDisplayNames: AutoReportingFrequencyDisplayNames = {
+    [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MONTHLY]: 'Monthly',
+    [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.IMMEDIATE]: 'Daily',
+    [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.WEEKLY]: 'Weekly',
+    [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.SEMI_MONTHLY]: 'Twice a month',
+    [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.TRIP]: 'By trip',
+    [CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MANUAL]: 'Manually',
+};
+
+const DAYS_OF_MONTH = 29;
+
+function WorkspaceAutoReportingFrequencyPage({policy}: WorkspaceAutoReportingFrequencyPageProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [isMonthlyFrequency, setIsMonthlyFrequency] = useState(policy?.autoReportingFrequency === CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MONTHLY);
+
+    const autoReportingFrequencyItems: WorkspaceAutoReportingFrequencyPageSectionItem[] = Object.keys(autoReportingFrequencyDisplayNames).map((frequencyKey) => {
+        const isSelected = policy?.autoReportingFrequency === frequencyKey;
+
+        return {
+            text: autoReportingFrequencyDisplayNames[frequencyKey as AutoReportingFrequencyKey] || '',
+            keyForList: frequencyKey,
+            isSelected,
+        };
+    });
+
+    const onSelectAutoReportingFrequency = (item: WorkspaceAutoReportingFrequencyPageSectionItem) => {
+        Policy.setWorkspaceAutoReportingFrequency(policy?.id ?? '', item.keyForList as AutoReportingFrequencyKey);
+        if (item.keyForList !== CONST.POLICY.AUTO_REPORTING_FREQUENCIES.MONTHLY) {
+            Navigation.goBack();
+        }
+        setIsMonthlyFrequency(true);
```

Comment:
This seems off since we're always setting `isMonthlyFrequency = true` even when the frequency is something else

Reply:
True true, I've modified it